tests/unit/amo/pages/TestAddonInfo.js

```diff
@@ -70,82 +77,351 @@ describe(__filename, () => {
+      _loadAddonResults([addon]);
+      _loadVersions({
         slug,
-      }),
-    );
-  });
+        versions: [
+          {
+            ...fakeVersion,
+            license: { ...fakeVersion.license, text: undefined },
+          },
+        ],
+      });
+      const errorHandler = createStubErrorHandler();
 
-  it('does not fetch addonInfo if it is already loaded', () => {
-    const slug = 'some-addon-slug';
-    _loadAddonInfo({ slug });
-    const errorHandler = createStubErrorHandler();
+      const dispatch = sinon.stub(store, 'dispatch');
 
-    const fakeDispatch = sinon.stub(store, 'dispatch');
+      render({
+        errorHandler,
+        infoType: ADDON_INFO_TYPE_CUSTOM_LICENSE,
+        params: { slug },
+      });
 
-    render({
-      errorHandler,
-      params: { slug },
+      sinon.assert.calledWith(
+        dispatch,
+        fetchVersion({
+          errorHandlerId: errorHandler.id,
+          slug,
+          versionId: addon.current_version.id,
+        }),
+      );
     });
 
-    sinon.assert.neverCalledWith(
-      fakeDispatch,
-      fetchAddonInfo({
-        errorHandlerId: errorHandler.id,
+    it('does not fetch an addonVersion when the loaded version has license text', () => {
+      const slug = 'some-addon-slug';
+      const addon = { ...fakeAddon, slug };
+      _loadAddonResults([addon]);
+      _loadVersions({
         slug,
-      }),
-    );
+        versions: [
+          {
+            ...fakeVersion,
+            license: { ...fakeVersion.license, text: 'some text' },
+          },
+        ],
+      });
+      const errorHandler = createStubErrorHandler();
+
+      const dispatch = sinon.stub(store, 'dispatch');
+
+      render({
+        errorHandler,
+        infoType: ADDON_INFO_TYPE_CUSTOM_LICENSE,
+        params: { slug },
+      });
+
+      sinon.assert.neverCalledWith(
+        dispatch,
+        fetchVersion({
+          errorHandlerId: errorHandler.id,
+          slug,
+          versionId: addon.current_version.id,
+        }),
+      );
+    });
+
+    it('does not fetch an addonVersion if one is already loading', () => {
+      const slug = 'some-addon-slug';
+      const addon = { ...fakeAddon, slug };
+      _loadAddonResults([addon]);
+      const errorHandler = createStubErrorHandler();
+
+      store.dispatch(
+        fetchVersion({
+          errorHandlerId: errorHandler.id,
+          slug,
+          versionId: addon.current_version.id,
+        }),
+      );
+
+      const dispatch = sinon.stub(store, 'dispatch');
+
+      render({
+        errorHandler,
+        infoType: ADDON_INFO_TYPE_CUSTOM_LICENSE,
+        params: { slug },
+      });
+
+      sinon.assert.neverCalledWith(
+        dispatch,
+        fetchVersion({
+          errorHandlerId: errorHandler.id,
+          slug,
+          versionId: addon.current_version.id,
+        }),
+      );
+    });
+
+    it('fetches an addonVersion when the slug changes', () => {
```

Comment:
I see in the above eula and privacy tests you combined addon and addonInfo for example. I am wondering if we could do the same with this one - maybe combine it with the test above (https://github.com/mozilla/addons-frontend/pull/7018/files#diff-1ed138d04a7c5e0e4ebdfe289ce4dafdR209)?

Reply:
This test is for when the slug changes, which I think is different from the test you pointed to. Also, the reason I combined them in the tests for eula and privacy policy is that the fetches are done in parallel for those, because each fetch is based on slug. In the case of the custom license, first a fetch needs to be done for the add-on, which loads the version behind the scenes, but without license text, and then the second fetch has to be done to get the version details using the versionId from the add-on. It seemed like the best way to test that would be to pre-load the add-on, and only test the second part, which is what I've done in this test.

Reply:
sorry about that. Yes I referenced the wrong line :/ 

I see what you're saying tho...thanks for explaining this